### PR TITLE
Get song duration, bitrate and codec from Pandora

### PR DESCRIPTION
**Pull request by JasonLG1979**
Get song duration, bitrate and codec from Pandora for use in the UI. It makes for a smoother UI experience IMHO as that info is instantly available as apposed to waiting for Pithos/Gstreamer to parse the data.

### CODE REVIEW

**Comment by JasonLG1979** [comment]
Combined with @TingPing's buffer spinner the UI will be like butter.

**Comment by TingPing** [comment]
You can get rid of `gst_tag_handler` then. Also you still make use of the duration info gst has, is that intended to mix and match the two?

File: pithos/pithos.py [hunk 1]
```diff
@@ -639,14 +641,14 @@ def on_gst_async_done(self, bus, message):
...
-        self.current_song.duration_message = self.format_time(self.current_song.duration)
+        self.current_song.ad_duration_message = self.format_time(self.current_song.duration)
         self.check_if_song_is_ad()
         self.player_status.pending_duration_query = False
 
     def on_gst_duration_changed(self, bus, message):
       if self.player_status.async_done:
         self.current_song.duration = self.query_duration()
-        self.current_song.duration_message = self.format_time(self.current_song.duration)
+        self.current_song.ad_duration_message = self.format_time(self.current_song.duration)
```
Comment (TingPing) [inline pithos/pithos.py]: Whats the purpose of ad_duration_message if its set to the same thing?

**Comment by JasonLG1979** [comment]
No we really don't need the tag handler anymore but it might not be a bad idea to keep it around for debug purposes?

We still need the gstreamer duration to tell if a song is an ad. Pandora lies about duration of ads to.

They're mixed in song_text because if it's an ad we want to use the gstreamer duration if not we can use the duration provided by Pandora.

**Comment by TingPing** [comment]
> We still need the gstreamer duration to tell if a song is an ad. Pandora lies about duration of ads to.

I think for now it makes more sense to just trust pandora on codec/bitrate, and trust gstreamer on duration. There is little point in mixing the two when they may not match.

**Comment by JasonLG1979** [comment]
Fair enough.

**Comment by JasonLG1979** [comment]
What about the tag handler? I say maybe keep it but make the debug message a little more useful/less spammy

**Comment by TingPing** [comment]
I don't see much value in it but that's fine. Maybe only hook that event if log level == DEBUG to improve performance, since that is sort of the point of these changes right.

**Comment by JasonLG1979** [comment]
After thinking about it I agree, the tag handler is pretty useless if we get the codec and bitrate from Pandora. It's a whole lot of code just to spam the debug messages.

I'm going to close this and start over. No duration from Pandora, just the bitrate and codec, and nix the tag handler. Sound good?

**Comment by TingPing** [comment]
> I'm going to close this and start over. No duration from Pandora, just the bitrate and codec, and nix the tag handler. Sound good?

Sounds good. It would be nice for you to look into `git` more deeply to avoid multiple PRs though ;)

**Comment by JasonLG1979** [comment]
Will do. It was just easier to start fresh than undo everything.

**Pull request closed without merging**